R/board_url.R

```diff
@@ -1,52 +1,95 @@
 #' Use a vector of URLs as a board
 #'
 #' @description
-#' `board_url()` lets you build up a board from individual urls. This is
-#' useful because [pin_download()] and [pin_read()] will be cached - they'll
-#' only re-download the data if it's changed from the last time you downloaded
-#' it (using the tools of
-#' [HTTP caching](https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching)).
-#' You'll also be protected from the vagaries of the internet; if a fresh
-#' download fails, you'll get the previously cached result with a warning.
+#' `board_url()` lets you build up a board from individual urls or a [manifest
+#' file][write_board_manifest()].
 #'
-#' `board_url()` is read only and does not currently support versions.
+#' `board_url()` is read only.
 #'
-#' @param urls A named character vector of URLs If the URL ends in a `/`,
-#'   `board_url` will look for a `data.txt` that provides metadata. The
-#'   easiest way to generate this file is to upload a pin directory created by
-#'   [board_folder()].
+#' @param urls Identify available pins being served at a URL or set of URLs (see details):
+#'   - Unnamed character scalar: URL to a [manifest file][write_board_manifest()].
+#'   - Named character vector: URLs to specific pins (does not support versioning).
+#'   - Named list: URLs to pin version directories (supports versioning).
 #' @param use_cache_on_failure If the pin fails to download, is it ok to
 #'   use the last cached version? Defaults to `is_interactive()` so you'll
 #'   be robust to poor internet connectivity when exploring interactively,
 #'   but you'll get clear errors when the code is deployed.
 #' @family boards
 #' @inheritParams new_board
+#' @details
+#' The way `board_url()` works depends on the type of the `urls` argument:
+#'   - Unnamed character scalar, i.e. **a single URL** to a
+#'     [manifest file][write_board_manifest()]: If the URL ends in a `/`,
+#'     `board_url()` will look for a `_pins.yaml` manifest. If the manifest
+#'     file parses to a named list, versioning is supported. If it parses to a
+#'     named character vector, the board will not support versioning.
```

Comment:
Why would you point it to something other than a directory?

Reply:
It is possible to pass in a URL to a file that is a manifest file but not named `_pins.yaml`, like this:

``` r
library(pins)
b1 <- board_folder(withr::local_tempdir(), versioned = TRUE)
b1 %>% pin_write(1:10, "x", type = "json")
#> Creating new version '20221215T205631Z-c3943'
#> Writing to pin 'x'
b1 %>% pin_write(11:20, "y", type = "json")
#> Creating new version '20221215T205631Z-cba09'
#> Writing to pin 'y'
b1 %>% pin_write(1:20, "y", type = "csv")
#> Creating new version '20221215T205631Z-5026d'
#> Writing to pin 'y'
write_board_manifest(b1)
#> Manifest file written to root folder of board, as `_pins.yaml`
fs::file_move(fs::path(b1$path, "_pins.yaml"), fs::path(b1$path, "_different.yaml"))
b1_path <- fs::path(b1$path)

b1_server <- webfakes::new_app()
b1_server$use(webfakes::mw_static(root = b1_path))
b1_process <- webfakes::new_app_process(b1_server)

b2 <- board_url(fs::path(b1_process$url(), "_different.yaml"))
b2 %>%
  pin_read("x") 
#> → <http:/127.0.0.1:64054/x/20221215T205631Z-c3943/data.txt> is not cacheable
#> → <http:/127.0.0.1:64054/x/20221215T205631Z-c3943/x.json> is not cacheable
#>  [1]  1  2  3  4  5  6  7  8  9 10
```

<sup>Created on 2022-12-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I'm sure it won't be the most common use case but seems like some nice flexibility to offer.